### ISSUE TITLE
Delete photos older/earlier than specified date

### ISSUE THREAD

**Issue by mariusa** [mrishab/google-photos-delete-tool#52]
Similar to `maxImageCount`, please have a variable to delete photos older/earlier than specified date. Thanks!

**Comment by simonmcnair** [comment]
I agree, I'd love to be able to delete all photos since google stopped free storage on june 1 2021

**Comment by phanex** [comment]
I agree too. Trying to find a way to delete all from Jun 2021, but don't found any. I thought, that rclone may help, but Google API doesn't allow to.

**Comment by mrishab** [comment]
## AI-Generated Summary and Implementation Plan

**Issue Analysis**: Users are requesting a feature to delete all photos older than a specified date (e.g., before June 1, 2021 when Google stopped free storage). This is a valuable feature that complements the existing `maxImageCount` parameter.

**Use Cases**:
- Delete all photos before free Google Photos storage ended (June 1, 2021)
- Bulk delete old photos from a specific date range
- Better control over deletion criteria beyond just quantity

**Proposed Implementation**:

1. **Add a new configuration parameter**: `minPhotoDate` or `deleteBeforeDate` (format: "YYYY-MM-DD" or timestamp)
   ```javascript
   const minPhotoDate = new Date('2021-06-01').getTime(); // or use 0 for oldest first
   ```

2. **Extract photo date**: Google Photos displays the photo date/time. Need to:
   - Parse the date from the photo metadata or UI element
   - Convert to comparable format
   - Compare with user-specified date threshold

3. **Filter logic**: 
   - Before selecting a photo for deletion, check if its date is older than the specified threshold
   - Skip photos newer than the threshold
   - Only select photos from before the specified date

4. **Suggested variable names and usage**:
   ```javascript
   const deleteBeforeDate = new Date('2021-06-01'); // Only delete older than this
   // OR
   const maxPhotoDate = new Date('2021-06-01'); // Alternative naming
   ```

**Implementation Steps**:
1. Identify how Google Photos exposes the photo date/timestamp in the DOM
2. Add date parsing logic
3. Modify the selection logic to compare dates
4. Add user parameter documentation
5. Test with various date formats and edge cases

**Technical Challenges**:
- Google Photos may not always display dates in a standardized format
- Date parsing may need to handle different locales/languages
- Need to ensure reliable date extraction from the UI

**Recommendation**: This is a worthwhile feature that would increase the tool's usability significantly. Priority: Medium-High

**Comment by shtse8** [comment]
Feature request noted. The maintained continuation (https://github.com/shtse8/Google-Photos-Delete-Tool) doesn't have a date filter yet either — it deletes everything or a fixed batch count. If you'd like this, opening a feature request there is the best place so it can actually get implemented.
